core/src/main/java/org/apache/iceberg/BaseMetastoreCatalog.java

```diff
@@ -70,12 +87,42 @@ public Table createTable(
   public Table loadTable(TableIdentifier identifier) {
     TableOperations ops = newTableOps(conf, identifier);
     if (ops.current() == null) {
-      throw new NoSuchTableException("Table does not exist: " + identifier);
+      String name = identifier.name();
+      TableType type = TableType.from(name);
```

Comment:
Does this prevent creating tables that match one of the catalog TableTypes?

Reply:
Yes, this would prevent creating a table named `db.table.history` if `db.table` exists and `db.table.history` is a valid table name. Usually this won't happen because namespaces are some fixed depth. Hive, for example, only supports a single identifier as a namespace, which guarantees that this would not happen.